addon/addon.go

```diff
@@ -70,7 +84,37 @@ func getValue(cluster *clusterv1.ManagedCluster,
 			},
 		},
 	}
+	if val, ok := addon.GetAnnotations()["addon.open-cluster-management.io/search_memory_limit"]; ok {
+		match, err := regexp.MatchString(resourceLimitReg, val)
+		if err != nil {
+			addonLog.Info("Error parsing memory limit for cluster %s", cluster.ClusterName)
+		}
+		if match {
```

Comment:
Maybe this should be `else if` to be sure it doesn't get set if there's an error.
```suggestion
		} else if match {
```